### PR TITLE
Move labels to the same device as logits for LlamaForSequenceClassification and Blip2

### PR DESCRIPTION
**Pull request by xssChauhan**
# What does this PR do?
Fixes issue #22561 by moving the labels to the same device as the logits they are compared to for `LlamaForSequenceClassification`, `Blip2`.

@sgugger Could you review this once?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for your PR but it seems there is an issue with your CircleCI permissions, the tests won't run.
Could you try refreshing your permissions as shown [here](https://support.circleci.com/hc/en-us/articles/360048210711-How-to-Refresh-User-Permissions-)?

**Comment by xssChauhan** [comment]
@sgugger  I have refreshed my permissions, but I still do not see the option of rerunning the pipeline on CircleCI. Is it possible that I cant do that?

**Comment by sgugger** [comment]
You can just push an empty commit: `git commit -m "Trigger CI" --allow-empty`

**Comment by xssChauhan** [comment]
@sgugger  Is such a long waiting time for CircleCI report expected?

**Comment by sgugger** [comment]
Could you try again? Tests still aren't run.

**Comment by sgugger** [comment]
Perfect, thanks!
